validator/src/journal/chain.rs

```diff
@@ -263,6 +266,19 @@ impl<BC: BlockCache + 'static, BV: BlockValidator + 'static, CW: ChainWriter + '
             let chain_head_block = new_block.clone();
             state.chain_head = Some(new_block);
 
+            state.state_prune_manager.update_queue(
+                &result
+                    .new_chain
+                    .iter()
```

Comment:
Would it make more sense to have the function take care of this `.iter.map.collect` stuff rather than make the caller do it?

Reply:
The state pruning manager operates on state root hashes, not blocks